### ISSUE TITLE
Resources File in .NET Standard (1.4) with New csproj

### ISSUE THREAD

**Issue by tarekgh** [Microsoft/msbuild#1984]
I'm developing an application targeting .NET Standard 1.4, using the new csproj style (not project.json), and trying to get a localizable resources file to become available throughout the project.

I get compiler errors when including a resources file currently, similar to 'Resources' does not contain a definition for 'X'

The csproj automatically includes the resources file as follows:

    <ItemGroup>
        <Compile Update="Resources.Designer.cs">
            <DesignTime>True</DesignTime>
            <AutoGen>True</AutoGen>
            <DependentUpon>Resources.resx</DependentUpon>
        </Compile>
    </ItemGroup>

    <ItemGroup>
        <EmbeddedResource Update="Resources.resx">
            <Generator>ResXFileCodeGenerator</Generator>
            <LastGenOutput>Resources.Designer.cs</LastGenOutput>
        </EmbeddedResource>
    </ItemGroup>
Did it miss a step in the generation? Why can't I access any of the members of my resources file throughout the assembly?

http://stackoverflow.com/questions/43454441/resources-file-in-net-standard-1-4-with-new-csproj

**Comment by tarekgh** [comment]
https://github.com/dotnet/corefx/issues/18490

**Comment by tarekgh** [comment]
CC @dagrooms52

**Comment by dagrooms52** [comment]
Thanks for moving this @tarekgh 
As you can see in the stackoverflow link](http://stackoverflow.com/questions/43454441/resources-file-in-net-standard-1-4-with-new-csproj), I am working around this by using `Resources.ResourceManager.GetString("MyResource")`.

Creating a new project and recreating this scenario caused an intellisense error, but not a build error, so I was able to run the code; it is a real build error in my main project however, so I'm stuck with the workaround.

**Comment by ViktorHofer** [comment]
I just tried it in a sample solution inside Visual Studio 2017 15.1 (26403.3) and it works normally as expected. I first get the intellisense error but after waiting for a few seconds (and maybe also opening the Resource.Designer.cs file) the error disappears. Seems fine

**Comment by rainersigwald** [comment]
@dagrooms52 Can you be more specific about what you're doing and the error you see?

Possibly related: https://github.com/dotnet/sdk/issues/895.

**Comment by dagrooms52** [comment]
Sure.
When my repo was targeting PCL, we would access resources as a property on the resource class, e.g. `Resources.Resource1`. Since upgrading the repo to .NET Standard, this causes a compiler error. The members of the `Resources.resx` file are not propagated out to the `Resources.Designer.cs` class to be used as compile-time fields or properties. The class itself is accessible, but the resource members in it are only accessible through `Resources.ResourceManager.GetString("Resource1")`. I would prefer the compile-time properties to be available (run-time vs. compile-time errors being the main reason).

Having `Resources.Resource1` gives `error CS0117: 'Resources' does not contain a definition for 'Resource1'` as a build error.

**Comment by ViktorHofer** [comment]
@dagrooms52 which version of VS17 are you using? As I wrote above, I don't get any errors. Do you have a repro solution?

**Comment by dagrooms52** [comment]
I'm currently using Version 4.6.01586. I have a separate solution giving me an intellisense error but not a build error. So either the repro is fragile, or my main project affects the repro in some mysterious way.

**Comment by ViktorHofer** [comment]
Weird behavior. If you create an empty sample solution (see below), are you still facing the error?

1. Create a netstandard1.4 solution inside Visual Studio 2017
2. Add a new resource via Add New Item --> General --> Resource, e.g. called Resource1.resx
3. Add a string called e.g. test1
4. Verify if Resource1.Designer.resx got updated with the new property.
5. Try to access the string via "Resource1.test1".

**Comment by dagrooms52** [comment]
Yes, this gives me an intellisense error but not a compiler error.

**Comment by dagrooms52** [comment]
That works after refreshing VS, I think my problem may originate from copying xml from a PCL resource file into this new resource file. Looks like I'll be copying them out by hand, maybe I'll write a generator if there are too many.
@tarekgh can close.

**Comment by ViktorHofer** [comment]
That's even more obscure. The ResX Schema version 2.0 hasn't changed from PCL to netstandard class libraries. I just created a PCL assembly with a resx and made a diff against that from netstandard and they are completely identical. Maybe your Resx file wasn't a 2.0 file?

**Comment by dagrooms52** [comment]
I used the same method of creation in both cases; thought it was weird that you can only make a resources file through `Add New Item` -> `Apple`-> `Resources File`. It works in a fresh solution with new members added to the designer, but not in an existing solution when pasting existing resources xml.

**Comment by dagrooms52** [comment]
Chalk this one up to a noob not knowing how resource files work - the C# file is generated **after editing** the .resx file, not as a build step. Since I pasted all the xml I wanted and didn't do any edits in the actual .resx file, I never got a C# file generated. Once I edit to add a test resource, it generates the full class.

**Comment by tarekgh** [comment]
closing it as it looks resolved.

**Comment by ViktorHofer** [comment]
@dagrooms52 that makes sense! Thanks for providing the cause and the solution for your issue. Someone might appreciate it when he/she finds the thread 👍

**Comment by siebigteroth** [comment]
@tarekgh Does occur in 15.3.0 in a 1.4 .NET standard PCL project. If you add entries to a .resx-file (new project), it sometimes works immediately after using the table and saving, sometimes via Add Resource > Add New String and saving. But most of the times it doesn't work and I have to restart Visual Studio to see the changes in intellisense. This also applies to deleting or modifying entries.

**Comment by tarekgh** [comment]
@rainersigwald could you please have a look? I'll reopen the issue till we have a resolution. Thanks.

**Comment by brianjlacy** [comment]
I _was_ just experiencing this issue in a .NET Standard 1.6 project: I used the Resources Designer UI to add a few new strings to an existing Resources.resx file. The strings were correctly added to the XML, but nothing I tried would cause Visual Studio to regenerate the Resources.Designer.cs file, so I couldn't actually use the resource strings anywhere in my project.

Then I went back to the UI, and I noticed that the Access Modifier said "(Custom)". Changed that to Public, and suddenly I was back in business.

**Comment by watfordgnf** [comment]
Ran into this with a .NET Standard 2.0 project on the latest VS 2k19 preview. I added a new resource file via the standard menus, and then copy+pasted the resources I wanted into the resx designer. Just like with @brianjlacy it said "(Custom)" for the access modifier and I had to change it to Internal to make it work. However, the odd bit was the resources did not show up and I had to re-copy+paste them into the resx file.
